tasks.py

```diff
@@ -157,42 +184,73 @@ def generate_packages(context):
     run_command(context, command)
 
 
+@task(
+    help={
+        "check": (
+            "If enabled, check for outdated dependencies in the poetry.lock file, "
+            "instead of generating a new one. (default: disabled)"
+        )
+    }
+)
+def lock(context, check=False):
+    """Generate poetry.lock inside the Nautobot container."""
+    run_command(context, f"poetry {'check' if check else 'lock --no-update'}")
+
+
 # ------------------------------------------------------------------------------
 # START / STOP / DEBUG
 # ------------------------------------------------------------------------------
-@task
-def debug(context):
-    """Start Nautobot and its dependencies in debug mode."""
-    print("Starting Nautobot in debug mode...")
-    docker_compose(context, "up")
+@task(help={"service": "If specified, only affect this service."})
+def debug(context, service=""):
+    """Start specified or all services and its dependencies in debug mode."""
+    print(f"Starting {service} in debug mode...")
+    docker_compose(context, "up", service=service)
 
 
 @task(help={"service": "If specified, only affect this service."})
-def start(context, service=None):
-    """Start Nautobot and its dependencies in detached mode."""
+def start(context, service=""):
+    """Start specified or all services and its dependencies in detached mode."""
     print("Starting Nautobot in detached mode...")
     docker_compose(context, "up --detach", service=service)
 
 
-@task
-def restart(context):
-    """Gracefully restart all containers."""
+@task(help={"service": "If specified, only affect this service."})
+def restart(context, service=""):
+    """Gracefully restart specified or all services."""
     print("Restarting Nautobot...")
-    docker_compose(context, "restart")
+    docker_compose(context, "restart", service=service)
 
 
-@task
-def stop(context):
-    """Stop Nautobot and its dependencies."""
+@task(help={"service": "If specified, only affect this service."})
+def stop(context, service=""):
+    """Stop specified or all services, if service is not specified, remove all containers."""
     print("Stopping Nautobot...")
-    docker_compose(context, "down")
+    docker_compose(context, "stop" if service else "down --remove-orphans", service=service)
```

Comment:
Not sure `invoke stop` command should run `docker compose down --remove-orphans`. Probably should just run `docker compose stop`. Submitted issue https://github.com/nautobot/cookiecutter-nautobot-app/issues/36 and PR https://github.com/nautobot/cookiecutter-nautobot-app/pull/37 in the cookiecutter

Reply:
I kept the original behavior to use `down`, when there is no service specified, i.e. `invoke stop` calls `docker compose down`. Added `--remove-orphans` only. When service is specified, it's not possible to call `docker compose down -- <service>` with single service, so `docker compose stop <service>` is used.

If we can drop backward compatibility, happy to fix this, to use `docker compose stop` only. WDYT @whitej6 ?

Reply:
Only issue I see is possible verifying we have a line in the dev env docs stating the minimum version of docker-cli if there's an incompatibility issue with other versions of docker-cli